dashboard/api_test.py

```diff
@@ -232,6 +232,34 @@ def test_format_run_with_currently_enrolled(self):
             self.expected_ret_data
         )
 
+    def test_format_run_currently_enrolled_dont_display_progress(self):
+        """
+        test that setting `should_display_progress` to False prevents
+        the current grade from being returned
+        """
+        self.expected_ret_data.update({
+            'status': api.CourseStatus.CURRENTLY_ENROLLED
+        })
+        del self.expected_ret_data['final_grade']
+        self.crun.course.should_display_progress = False
+        self.assertEqual(
+            api.format_courserun_for_dashboard(self.crun, api.CourseStatus.CURRENTLY_ENROLLED, self.mmtrack),
+            self.expected_ret_data
+        )
+        self.crun.course.should_display_progress = True
```

Comment:
Instead of resetting the value manually can you add `self.course.refresh_from_db()` to `setUp` before `create_run` is called?